kmk/modules/combos.py

```diff
@@ -223,6 +226,9 @@ def on_release(self, keyboard, key, int_coord):
                 self._key_buffer.append((int_coord, key, False))
                 key = None
 
+        # Reset on non-combo key up
+        if not self._matching:
+            self.reset(keyboard)
```

Comment:
This obviously only resets if _no_ combos are matching, and doesn't reset non-matching combos if there're potential matches left. I'm not 100% sure that's perfect, but it's at least a step foward.